sklearn/manifold/tests/test_t_sne.py

```diff
@@ -110,6 +116,39 @@ def test_binary_search():
     assert_almost_equal(mean_perplexity, desired_perplexity, decimal=3)
 
 
+def test_binary_search_neighbors():
+    """Test if the binary perplexity search is approximately equal to the
+       slow method when the slow method uses all points as nighbors"""
+    n_samples = 500
+    desired_perplexity = 25.0
+    random_state = check_random_state(0)
+    distances = random_state.randn(n_samples, 2)
+    # Distances shouldn't be negative
+    distances = np.abs(distances.dot(distances.T))
+    np.fill_diagonal(distances, 0.0)
+    P1 = _binary_search_perplexity(distances, desired_perplexity,
+                                   verbose=0)
+
+    # Test that when we use all the neighbors the results are identical
+    k = n_samples
+    neighbors_nn = np.argsort(distances, axis=1)[:, :k]
+    P2 = _binary_search_perplexity_nn(distances, neighbors_nn,
+                                      desired_perplexity, verbose=0)
+    assert_array_almost_equal(P1, P2, decimal=6)
+
+    # Test that the highest P_ij are the same when few neighbors are used
+    for k in np.linspace(80, n_samples, 10):
+        k = int(k)
+        topn = k * 10  # check the top 10 *k entries out of k * k entries
+        neighbors_nn = np.argsort(distances, axis=1)[:, :k]
+        P2k = _binary_search_perplexity_nn(distances, neighbors_nn,
+                                           desired_perplexity, verbose=0)
+        idx = np.argsort(P1.ravel())[::-1]
+        P1top = P1.ravel()[idx][:topn]
+        P2top = P2k.ravel()[idx][:topn]
+        assert_array_almost_equal(P1top, P2top, decimal=2)
+
```

Comment:
Maybe it would be interesting to add a test checks that:

`TSNE(method='standard')` yields an embedding with a perplexity or 1-NN query accuracy similar to one generated with `TSNE(method='barnes_hut', angle=0)`.